enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,605 @@
+
+* We will not offer support for custom labeling of logs other than the defaults provided by the cluster logging collector.
+
+## Proposal
+
+To offer a seamless cluster logging experience with Loki as an alternative store, we propose to limit the API support to the following key features:
+
+* Deploy Loki **only** in microservices mode, i.e. each component is deployed and managed as an individual deployment/statefulset. This enables individual capacity planning and resource utilization per path ingestion/querying.
+
+* Multi-tenancy is enabled by default and requires the collector to apply the required [tenant identification](https://grafana.com/docs/loki/latest/operations/multi-tenancy/) on ingestion time. No user can disable this setting. In turn queries require the same identification information to be successful. Furthermore there is no support for querying across tenants.
+
+* Loki is operated only in a zero-dependency setup mode as described [here](https://grafana.com/docs/loki/latest/configuration/examples/#almost-zero-dependencies-setup). The only externally managed requirement is the availability of an S3-compatible API for object storage to persist logs.
+
+* Support a limited set of Loki scale outs, which are battle-tested on our internally managed Loki deployment. The target scale outs are defined using the format `Nx.{extra-small, small, medium}` where `N` the number of Loki deployments:
+  * **1x.extra-small**: Dedicated scale for development and demo purposes only! Not a valid configuration for cluster logging production environments.
+  * **1x.small**: Dedicated scale for low-retention log storage without replication factor support. All components are deployed with HA by running at least two replicas.
+  * **1x.medium**: Dedicated scale for log-retention log storage with single replication factor support. All components are deployed with HA by running at least two replicas. The only exception is the ingester that requires at minimum three replicas to allow failover when single replication factor enabled.
+
+* Support tuning of CPU/Memory resource and limits for each individual component beyond the selected scale-out size.
+
+* Support adding kubernetes tolerations and node selectors for all Loki components individually. This enables operating the Loki deployment on dedicated nodes.
+
+* Support a limited set of control options from Loki's [limit_config](https://grafana.com/docs/loki/latest/configuration/#limits_config) for the ingestion and query paths. The set is an opinionated list of options to enable tuning of object storage usage. The list of options is offered globally across the whole Loki deployment or per tenant.
+
+For **1x.medium** instances only:
+
+* Support optionally replication factor of two for log store on the target object storage.
+
+* Support auto-compaction of the index and chunks stored on the object storage when replication factor of two enabled.
```

Comment:
didn't we discuss we would have this for anything with RF2, so compaction should also apply for `1x.small` right?